src/core/Akka.TestKit/EventFilter/Internal/EventFilterApplier.cs

```diff
@@ -442,7 +442,23 @@ public EventFilterFactory And
                 ? TestKitExtension.For(system).TestEventFilterLeeway
                 : _testkit.TestKitSettings.TestEventFilterLeeway;
 
-            var timeoutValue = timeout.HasValue ? _testkit.Dilated(timeout.Value) : leeway;
+            TimeSpan timeoutValue;
+            if (timeout.HasValue)
+            {
+                timeoutValue = _testkit.Dilated(timeout.Value);
+            }
+            else
+            {
+                timeoutValue = _testkit.RemainingOrDilated(leeway);
+                // If we expected 0 occurence and we're inside a Within block, timeout need to be slightly less else Within will throw
+                if (_testkit.IsInsideWithin && expectedOccurrences == 0)
```

Comment:
Special check when occurance is 0 and we're within a Within block. timeout need to be slightly less or Within WILL throw